src/domain/messages/entities/schemas/__tests__/message.schema.spec.ts

```diff
@@ -0,0 +1,247 @@
+
+        expect(!result.success && result.error).toStrictEqual(
+          new ZodError([
+            {
+              code: 'custom',
+              message: 'Invalid input',
+              path: [key],
+            },
+          ]),
+        );
+      },
+    );
+
+    it.each([
+      ['string', faker.lorem.sentence()],
+      ['object', JSON.parse(fakeJson())],
+    ])('should allow a %s message', (_, message) => {
+      const result = MessageSchema.safeParse({
+        ...messageBuilder().build(),
+        message,
+      });
+
+      expect(result.success).toBe(true);
+    });
+
+    it.each(['safeAppId' as const, 'preparedSignature' as const])(
+      'should allow undefined %s, defaulting to null',
+      (key) => {
+        const message = messageBuilder().build();
+        delete message[key];
+
+        const result = MessageSchema.safeParse(message);
+
+        expect(result.success && result.data[key]).toBe(null);
+      },
+    );
+
+    it('should allow empty confirmations', () => {
+      const message = messageBuilder().with('confirmations', []).build();
+
+      const result = MessageSchema.safeParse(message);
+
+      expect(result.success).toBe(true);
+    });
+
+    it.each([
+      'created' as const,
+      'modified' as const,
+      'safe' as const,
+      'messageHash' as const,
+      'message' as const,
+      'proposedBy' as const,
+      'confirmations' as const,
+    ])('should not allow %s to be undefined', (key) => {
+      const message = messageBuilder().build();
+      delete message[key];
+
+      const result = MessageSchema.safeParse(message);
+
+      expect(
+        !result.success &&
+          result.error.issues.length === 1 &&
+          result.error.issues[0].path.length === 1 &&
+          result.error.issues[0].path[0] === key,
+      ).toBe(true);
+    });
+  });
+
+  describe('MessagePageSchema', () => {
+    it('should validate a valid Page<Message>', () => {
+      const message = messageBuilder().build();
+      const messagePage = pageBuilder().with('results', [message]).build();
+
+      const result = MessagePageSchema.safeParse(messagePage);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should allow empty pages', () => {
+      const messagePage = pageBuilder().with('results', []).build();
+
+      const result = MessagePageSchema.safeParse(messagePage);
+
+      expect(result.success).toBe(true);
+    });
+
+    it.each<keyof Page<Message>>(['count', 'next', 'previous', 'results'])(
+      'should not allow %s to be undefined',
+      (key) => {
+        const messagePage = pageBuilder().with('results', []).build();
+        delete messagePage[key];
+
+        const result = MessagePageSchema.safeParse(messagePage);
+
+        expect(
+          !result.success &&
+            result.error.issues.length === 1 &&
+            result.error.issues[0].path.length === 1 &&
+            result.error.issues[0].path[0] === key,
+        ).toBe(true);
+      },
+    );
+  });
```

Comment:
Nit: these two tests are independent of the type of the items in the Page, what about extracting them to a `page.schema.spec.ts`? Otherwise we would be repeating them for all the `Page<T>` types.

This is not an strong opinion btw, and it would also require to export `PageSchema` somewhere as well, so I think this can be tackled in a separate PR if you agree with the suggestion.

Reply:
It's a valid point, which I agree with. I removed them in 1a939c76af3ca4fbea86456f98c2d9428aa2758f and will create a follow up PR for the `PageSchema`.

Reply:
Great, thanks!